### ISSUE TITLE
Confusion surrounding mdl-layout--overlay-drawer-button in Layout documentation 

### ISSUE THREAD

**Issue by SeanBannister** [google/material-design-lite#1289]
Apologies if I don't understand this correctly.

In the [Layout documentation](http://www.getmdl.io/components/index.html#layout-section) the description for the ```mdl-layout--overlay-drawer-button``` class is "Defines layout as having a button-activated drawer" but some of the codepen examples on that page have a hamburger icon that opens the draw but they don't use the ```mdl-layout--overlay-drawer-button``` class. I'm wondering if these are incorrect or maybe the description for ```mdl-layout--overlay-drawer-button``` should be clearer in describing what its used for.

**Comment by eduardoboucas** [comment]
I too have some questions around this. In the configuration options table, `mdl-layout--overlay-drawer-button` is referred to as being *Required on outer div element*, however the button is still displayed when the class is absent.
Also, `mdl-layout-icon` is also said to be *Required on icon div*, but apparently it's not required.

I'm trying to create a layout without a header, just a drawer (collapsible). Here's what I have:

```html
<div class="mdl-layout mdl-js-layout">
  <div class="mdl-layout__drawer">
    <span class="mdl-layout-title">Title</span>
    <nav class="mdl-navigation">
      <a class="mdl-navigation__link" href="">Link</a>
      <a class="mdl-navigation__link" href="">Link</a>
      <a class="mdl-navigation__link" href="">Link</a>
      <a class="mdl-navigation__link" href="">Link</a>
    </nav>
  </div>
  <main class="mdl-layout__content">
    <div class="page-content">
        <p>Content here!</p>
    </div>
  </main>
</div>
```

The result of that is, in fact, a collapsible drawer and no header. But a hamburger icon is being added (`div.mdl-layout__drawer-button`) without me specifying anywhere in the markup that I want this button. Is this expected behaviour? Is it because the drawer element necessarily needs an icon so it just adds it?

In this case it's absolutely fine for me, because I do want the element, but I worry that this is not expected behaviour and might change in future versions. It'd be great if some collaborator could clarify.

Many thanks.

**Comment by surma** [comment]
Thanks opening this issue! Apparently, we missed some update on the docs here, as the `mdl-layout--overlay-drawer-button` doesn’t even exist (anymore) in [_layout.scss](https://github.com/google/material-design-lite/blob/4890f013132c71d4686da0508787d81fa01eb215/src/layout/_layout.scss).

@sgomes Can you confirm that the class is outdated and can be removed from the docs?

**Comment by sgomes** [comment]
Yup, that class has been removed, and we should update the docs. Good catch, @SeanBannister!

As for the button being added automatically in your situation, @eduardoboucas, that is expected behaviour in the 1.x range and won't change until an eventual 2.x, so feel free to rely on it. We may add a class to disable it with a 1.x minor release, but the default behaviour will remain the current one.

**Comment by eduardoboucas** [comment]
Brilliant, thanks guys!

**Comment by SeanBannister** [comment]
Awesome :) I just checked all documented layout classes to ensure there weren't any others that should be removed, everything else looks good.

In regards to @eduardoboucas question regarding ```mdl-layout-icon``` I also noticed an inconsistency, I've posted #1294 to discuss further.

**Comment by eduardoboucas** [comment]
@sgomes For what it's worth, here's my opinion about that behaviour. I've never used any UI libraries in production, particularly because I'm not comfortable with the idea of not being in control of the markup and the styles introduced in the page (not to mention that some of the libraries available out there have poorly implemented components). I'm spending some time experimenting with MDL because it's obviously built properly by very smart people, with standards and best practices in mind, so I wanted to give it a try. 

I'm reading the documentation thoroughly to understand exactly what each class does in hopes of alleviating the concern I mentioned above, and the fact that an element is automatically being added to the markup without notice or information in the documentation doesn't really help.

I understand that the nature of HTML/CSS/JS makes it difficult to implement complex elements like this in a transparent and seamless way, but in my humble opinion the components should try to avoid making decisions/assumptions for the user as much as possible — even if that means asking people to add an extra class or element if they want that hamburger button, which will probably be the case for 90% of the people.

**Comment by surma** [comment]
We have had similar feedback and reducing JavaScript black magic is one of the major goals vor v2. For now, it would be a breaking change we can’t really change this in the v1.x timeline.

**Comment by sgomes** [comment]
@eduardoboucas Thank you for the detailed writeup and for the kind words! :)

Like @surma said, automatic element creation is something we definitely want to reduce as much as possible in v2, if not remove altogether. The drawer button and a few other layout bits are perfect examples of this, where we could probably do with a better balance between amount of boilerplate and maintainability/debuggability.

As for documentation, layout is probably in need of a [thorough review of its docs](https://github.com/google/material-design-lite/issues/1310). It could also benefit from extra documentation, such as a more task-driven, ["getting started" set of guides](https://github.com/google/material-design-lite/issues/1311), since it's the first thing developers have to deal with. I'll be working on both of those, so if there's anything specific you'd like to see, let me know!

**Comment by eduardoboucas** [comment]
@sgomes Thanks for that. I think that the guide you described in #1311 is a brilliant idea, I reckon it would help developers a lot. 

I'll keep experimenting with the library and comparing it against the documentation, and I'll flag anything that doesn't seem clear or consistent. If there's any other way I can help on my end, please let me know. 

Thanks!

**Comment by sgomes** [comment]
The original issue this refers to (bad docs for `mdl-layout--overlay-drawer-button`) should be fixed with the merging of #1327.

**Comment by SeanBannister** [comment]
Fixed in #1320
